webhooks.go

```diff
@@ -102,6 +142,47 @@ type CreateWebhookRequest struct {
 	FolderID string         `json:"folder_id,omitempty"`
 }
 
+type webhookVerifyResult struct {
+	validSignature       bool
+	signatureFromClickup string
+	signatureGenerated   string
+}
+
+// VerifyWebhookSignature compares a generated signature using secret that
+// is returned with the webhook CRUD operations with the x-signature http header
+// that is sent with the http request to the webhook endpoint.
+// It should be noted that err will be nil even if the signature is not valid,
+// thus the WebhookVerifyResult.ValidSignature() should be called.
+func VerifyWebhookSignature(webhookRequest *http.Request, secret string) (*webhookVerifyResult, error) {
```

Comment:
Chose to make the `webhookVerifyResult` unexported so that this method is the only way to create one.